lib/application.js

```diff
@@ -300,22 +300,34 @@ app.engine = function(ext, fn){
  * See the Router#param() docs for more details.
  *
  * @param {String|Array} name
- * @param {Function} fn
+ * @param {Function|Array} fn
  * @return {app} for chaining
  * @api public
  */
 
 app.param = function(name, fn){
+  var args = slice.apply(arguments);
+  var name = args[0];
+  var fns = args.slice(1);
+
   this.lazyrouter();
 
   if (Array.isArray(name)) {
     name.forEach(function(key) {
-      this.param(key, fn);
+      var params = [key].concat(fns);
+      this.param.apply(this, params);
     }, this);
     return this;
   }
 
-  this._router.param(name, fn);
+  if (fns.length == 0) {
```

Comment:
It is for backward compatibility with the old `app.param(function(){})` if I remember correctly